client/src/components/QualityManagement/AddDefectModal.jsx

```diff
@@ -104,4 +105,4 @@ AddDefectModal.propTypes = {
   showButton: PropTypes.bool,
 };
 
-export default AddDefectModal;
+export default observer(AddDefectModal);
```

Comment:
Why is using `observer` here and the other files necessary?

Reply:
because I am reading an observable state in the userStore

Reply:
because the component is reading an observable state in userStore

Reply:
Ahh yeah I see it now, my bad